frontend/src/pages/RegisterPage.js

```diff
@@ -44,16 +44,48 @@ export default function RegisterPage({ setIsLoggedIn }) {
     if (registerCredentials.password !== "") {
       checkPassword(registerCredentials.password);
     }
-    if (
-      registerCredentials.passwordChecked !== "" &&
-      registerCredentials.passwordChecked === registerCredentials.password
-    ) {
-      setPasswordCheckIsNotTheSameAsPassword(false);
-    } else {
-      setPasswordCheckIsNotTheSameAsPassword(true);
+    if (registerCredentials.passwordChecked !== "") {
+      if (
+        registerCredentials.passwordChecked !== "" &&
+        registerCredentials.passwordChecked === registerCredentials.password
+      ) {
+        setPasswordCheckIsNotTheSameAsPassword(false);
+      } else {
+        setPasswordCheckIsNotTheSameAsPassword(true);
+      }
     }
   }, [registerCredentials]); //eslint-disable-line react-hooks/exhaustive-deps
 
+  useEffect(() => {
+    if (usernameAlreadyRegistered !== undefined) {
+      if (usernameAlreadyRegistered === true) {
```

Comment:
```suggestion
    if (usernameAlreadyRegistered) {
      if (usernameAlreadyRegistered) {
```

Reply:
It works, if I change only the second variable. The first one needs to stay as "!== undefined" for my function to work properly :)